src/lib/sql-export/export-command.js

```diff
@@ -0,0 +1,329 @@
+	mutation GenerateDBBackupCopyUrl($input: AppEnvironmentGenerateDBBackupCopyUrlInput) {
+		generateDBBackupCopyUrl(input: $input) {
+			url
+			success
+		}
+	}
+`;
+
+const CREATE_EXPORT_JOB_MUTATION = gql`
+	mutation BackupDBCopy($input: AppEnvironmentStartDBBackupCopyInput) {
+		startDBBackupCopy(input: $input) {
+			message
+			success
+		}
+	}
+`;
+
+/**
+ * Fetches the latest backup and job status for an environment
+ * @param {int} appId Application ID
+ * @param {int} envId Environment ID
+ * @resolves {Promise} A promise which resolves to the latest backup and job status
+ */
+async function fetchLatestBackupAndJobStatus( appId, envId ) {
+	const api = await API();
+
+	const response = await api.query( {
+		query: BACKUP_AND_JOB_STATUS_QUERY,
+		variables: { appId, envId },
+		fetchPolicy: 'network-only',
+	} );
+
+	const {
+		data: {
+			app: { environments },
+		},
+	} = response;
+
+	const latestBackup = environments[ 0 ].latestBackup;
+	const jobs = environments[ 0 ].jobs;
+
+	return { latestBackup, jobs };
+}
+
+/**
+ * Generates a download link for a backup
+ * @param {int} appId Application ID
+ * @param {int} envId Environment ID
+ * @param {int} backupId Backup ID
+ * @resolves {Promise} A promise which resolves to the download link
+ */
+async function generateDownloadLink( appId, envId, backupId ) {
+	const api = await API();
+	const response = await api.mutate( {
+		mutation: GENERATE_DOWNLOAD_LINK_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				environmentId: envId,
+				backupId,
+			},
+		},
+	} );
+
+	const {
+		data: {
+			generateDBBackupCopyUrl: { url },
+		},
+	} = response;
+
+	return url;
+}
+
+/**
+ * Creates an export job for a backup
+ * @param {int} appId Application ID
+ * @param {int} envId Environment ID
+ * @param {int} backupId Backup ID
+ * @resolves {Promise} A promise which resolves to null if job creation succeeds
+ * @throws {Error} Throws an error if the job creation fails
+ */
+async function createExportJob( appId, envId, backupId ) {
+	const api = await API();
+	const response = await api.mutate( {
+		mutation: CREATE_EXPORT_JOB_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				environmentId: envId,
+				backupId,
+			},
+		},
+	} );
+
+	const {
+		data: {
+			startDBBackupCopy: { success },
+		},
+	} = response;
+
+	if ( ! success ) {
+		throw new Error();
+	}
+}
+
+/**
+ * Class representing an export command workflow
+ */
+export class ExportCommand {
+	app;
+	env;
+	downloadLink;
+	progressTracker;
+	outputFile;
+
+	/**
+	 * Creates an instance of ExportCommand
+	 * @param {object} app The application object
+	 * @param {object} env The environment object
+	 * @param {string} outputFile The output file path
+	 * @constructor
+	 */
+	constructor( app, env, outputFile ) {
+		this.app = app;
+		this.env = env;
+		this.outputFile = outputFile;
+		this.progressTracker = new ProgressTracker( [
+			{ id: 'prepare', name: 'Preparing' },
```

Comment:
nit: Maybe we should use constants for the id? i.e. `const PROGRESS_PREPARE = 'prepare'` defined somewhere in this file. Helps future readers that aren't familiar with `ProgressTracker` and minimize typo chances.